client/src/components/SearchBar/SearchBar.tsx

```diff
@@ -0,0 +1,47 @@
+import { useState, useEffect, useRef } from 'react'
+import './SearchBar.css'
+import { motion, AnimatePresence } from 'framer-motion'
+import { RxMagnifyingGlass } from 'react-icons/rx'
+
+export default function SearchBar() {
+
+  const [query, setQuery] = useState('');
+  const [lastScroll, setLastScroll] = useState('up');
+
+
+  //hides scrollbar when scrolling down, shows when scrolling up
+  useEffect(() => {
+    let waiting = false;
+    let scrollPos = 0;
+    function handleScroll(){
+      if (!waiting) {
+        let newScrollPos = window.scrollY;
+        newScrollPos > scrollPos ? setLastScroll("down") : setLastScroll("up");
+        scrollPos = newScrollPos;
+        waiting = true;
+        setTimeout(() => {waiting = false}, 100);
+      }
+    }
+    window.addEventListener('scroll', handleScroll);
+    return (() => window.removeEventListener('scroll', handleScroll));
+  },[])
+
+  function search() {
+    if (query === '') return;
+    //search for posts with passed tag
+  }
+
+  return (
+
+    <AnimatePresence>
+      {lastScroll == "up" &&
+        <motion.div className="search shadow" key="nav-menu" initial={{top: "-8rem"}} animate={{ top: "0"}} exit={{top: "-8rem"}}>
+            <input onChange={e => setQuery(e.target.value)} type="text" />
```

Comment:
It's a bit weird to have internal state set like this, but the contents of the input not controlled by the `query` variable. If ever someone calls `setQuery` to, for example, trim whitespace, it puts you in a state where your version of the text no longer matches what the input displays.  Either pass this in via `<input value={query} ...>`, or get rid of the query useState entirely, and just grab a ref to this control, and read the value as needed.

Reply:
Overlooked this. Fixed now!